Unite.Composer/Data/Datasets/DatasetsService.cs

```diff
@@ -0,0 +1,23 @@
+using Unite.Cache.Configuration.Options;
+using Unite.Composer.Data.Datasets.Models;
+
+namespace Unite.Composer.Data.Datasets;
+
+public class DatasetsService
+{
+    private readonly Repositories.DatasetsRepository _datasetsRepository;
+
+    public DatasetsService(IMongoOptions options)
+	{
+		_datasetsRepository = new Repositories.DatasetsRepository(options);
+	}
+
+	public async Task<IEnumerable<DatasetModel>> Load(DatasetModel data)
+	{
+		var datasets = (await _datasetsRepository.
+		WhereAsync(item =>item.Document.UserID == data.UserID && item.Document.Domain == data.Domain))
```

Comment:
I think it's a bit inconsistent, that:
**Id** as part of the document model is "Id" document.Id / Item.Id
but it's "ID" is part of the user id property "UserID".

I think in our system we use form "Id" for identifier, not "ID",
So I'd suggest to change that from "UserI**D**" to "UserI**d**".

Reply:
Generally it's a bit harder to understand, what happens with await block and why is it wrapped with "()".

You probably wanted to write it all in one line, but I'd personally separate it to two calls:

var datasets = await _datasetsRepository.WhereAsync(item =>item.Document.UserID == data.UserID && item.Document.Domain == data.Domain));

return datasets.Select(item => item.Document with { Id = item.id });